azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/aad/AADAuthenticationProperties.java

```diff
@@ -24,13 +24,13 @@
     private String environment;
     /**
      * Registered application ID in Azure AD.
+     * @NotEmpty only for SPA implicit mode
      */
-    @NotEmpty
     private String clientId;
     /**
      * API Access Key of the registered application.
+     * @NotEmpty only for SPA implicit mode
```

Comment:
same as above. SPA implicit mode looks confusing.

Reply:
I think I will change to front end implicit grant flow, actually, it's different with Spring OAuth2 implicit grant flow, as the OAuth2 implicit grant flow can also happen in the backend. Which is do authentication with implicit grant type in the backend.